sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/TestPubsubSignal.java

```diff
@@ -254,17 +254,19 @@ public POutput expand(PCollection<? extends T> input) {
     private SerializableFunction<Set<T>, Boolean> successPredicate;
     // keep all events seen so far in the state cell
 
-    @StateId("seenEvents")
-    private final StateSpec<SetState<T>> seenEvents;
+    private static final String SEEN_EVENTS = "seenEvents";
+
+    @StateId(SEEN_EVENTS)
+    private final StateSpec<BagState<T>> seenEvents;
 
     StatefulPredicateCheck(Coder<T> coder, SerializableFunction<Set<T>, Boolean> successPredicate) {
-      this.seenEvents = StateSpecs.set(coder);
+      this.seenEvents = StateSpecs.bag(coder);
```

Comment:
FYI @akedin the only reason to use `SetState` is if you want to do efficient membership checking. Since the code here reads the whole thing, it doesn't add any functionality. And more runners support `BagState` so I switched it.